app/models/plage_ouverture.rb

```diff
@@ -21,6 +21,19 @@ class PlageOuverture < ApplicationRecord
 
   has_many :webhook_endpoints, through: :organisation
 
+  scope :for_lieu, ->(lieu) { where(lieu: lieu) }
+  scope :for_motif, ->(motif_name, organisation_id) { joins(:motifs).where(motifs: { name: motif_name, organisation_id: organisation_id }) }
+  scope :in, lambda { |range|
+    return all if range.nil?
+
+    # first_day in range
+    # or
+    # recurrence present ending in range
+    # recurrence present ending after end range and start before start range
+    # recurrence present without end and starting before end range
+    where(first_day: range).or(where.not(recurrence: nil).where("tsrange(first_day, recurrence_until, '[)') && tsrange(?, ?)", range.begin.to_date, range.end.to_date))
```

Comment:
Je ferais bien une construction en deux temps: 
```suggestion
    not_recurring_in_range = where(recurrence: nil).where(first_day: range)
    recurring_in_range = where.not(recurrence: nil).where("tsrange(first_day, recurrence_until, '[)') && tsrange(?, ?)", range.begin.to_date, range.end.to_date)
    
    not_recurring_in_range.or(recurring_in_range)
```